app/src/main/java/pm/gnosis/heimdall/ui/two_factor/Select2FaActivity.kt

```diff
@@ -38,16 +50,41 @@ open class Select2FaActivity : NfcActivity() {
     }
 
     private fun initKeyCardViews() {
-        val nfcAvailable = NfcAdapter.getDefaultAdapter(this)?.isEnabled == true
-        onSelected(if (nfcAvailable) AuthenticatorInfo.Type.KEYCARD else AuthenticatorInfo.Type.EXTENSION)
-        if (nfcAvailable) select_authenticator_keycard_background.setOnClickListener { onSelected(AuthenticatorInfo.Type.KEYCARD) }
+        nfcAvailable = NfcAdapter.getDefaultAdapter(this) != null
+        onSelected(AuthenticatorInfo.Type.KEYCARD)
+        if (nfcAvailable) {
+            select_authenticator_keycard_background.setOnClickListener { onSelected(AuthenticatorInfo.Type.KEYCARD) }
+        } else {
+            select_authenticator_nfc_required.visible(true)
+            select_authenticator_keycard_nfc.visible(true)
+            nfc_required_get_in_touch.setOnClickListener {
+                startActivity(GetInTouchActivity.newIntent(this))
+            }
+        }
+
+        select_authenticator_keycard_description.apply {
+            val linkDrawable = ContextCompat.getDrawable(context, R.drawable.ic_external_link)!!
+            linkDrawable.setBounds(0, 0, linkDrawable.intrinsicWidth, linkDrawable.intrinsicHeight)
+            text = SpannableStringBuilder()
+                .appendText(getString(R.string.status_keycard_description),
+                    ForegroundColorSpan(Color.argb(if(nfcAvailable) 255 else 99, Color.red(currentTextColor), Color.green(currentTextColor), Color.blue(currentTextColor))))
```

Comment:
Can we please put colors into the resources and load them from there

Reply:
this changes alpha of the text span. "Learn more" remains always opaque, but description changes transparency if an option is not available. I would extract alpha value to res/values

Reply:
But they are just 2 different colors, just put them both in the resource file, right?